Asignment7/baseball_stats.sql

```diff
@@ -0,0 +1,485 @@
+-- Using the baseball database
+USE baseball;
+
+-- Show all the tables in the baseball database
+SHOW TABLES;
+
+-- appending the local date to the team_batting_stats and team_pitching_stats and storing in a table
+
+DROP TABLE IF EXISTS s_team_batting_stats;
+CREATE TABLE s_team_batting_stats
+    (INDEX Game_id_Ind(game_id),INDEX Team_id_ind(team_id), INDEX Local_date(local_date))
+        SELECT team_batting_counts.game_id as game_id,
+               team_batting_counts.team_id as team_id,
+               DATE(game.local_date) as local_date,
+               team_batting_counts.atBat as b_AtBat,
+               team_batting_counts.Hit as b_Hit,
+               team_batting_counts.finalScore as Runs,
+               team_batting_counts.Home_Run as b_Home_Run,
+               team_batting_counts.Sac_Fly as b_Sac_Fly,
+               team_batting_counts.Strikeout as b_StrikeOut,
+               team_batting_counts.Walk as b_Walk,
+               team_batting_counts.Hit_By_Pitch as b_Hit_by_Pitch,
+               team_batting_counts.plateApperance as b_plateApperance
+               FROM team_batting_counts
+        JOIN game
+        ON game.game_id = team_batting_counts.game_id;
+
+
+DROP TABLE IF EXISTS s_team_pitching_stats;
+CREATE TABLE s_team_pitching_stats
+    (INDEX Game_id_ind(game_id),INDEX Team_id_ind(team_id), INDEX Local_Date(local_date))
+    SELECT team_pitching_counts.game_id as game_id,
+               team_pitching_counts.team_id as team_id,
+               DATE(game.local_date) as local_date,
+               team_pitching_counts.atBat as p_AtBat,
+               team_pitching_counts.Hit as p_Hit,
+               team_pitching_counts.finalScore as Runs,
+               team_pitching_counts.Home_Run as p_Home_Run,
+               team_pitching_counts.Sac_Fly as p_Sac_Fly,
+               team_pitching_counts.Strikeout as p_StrikeOut,
+               team_pitching_counts.Walk as p_Walk,
+               team_pitching_counts.plateApperance as p_plateApperance
+               FROM team_pitching_counts
+        JOIN game
+        ON game.game_id = team_pitching_counts.game_id;
+
+-- calculating the rolling batting average of each team of 200 days
+
+DROP TABLE IF EXISTS s_team_bat_rolling_avg;
+CREATE TABLE s_team_bat_rolling_avg
+    (INDEX Game_id_IND(game_id),INDEX Team_id_IND(team_id), INDEX t_rolling_batting_average(t_rolling_battingaverage))
```

Comment:
You can use ENGINE=MEMORY before SELECT to calculate the rolling average even more faster.